discovery/file/file.go

```diff
@@ -256,6 +256,11 @@ func readFile(filename string) ([]*config.TargetGroup, error) {
 	}
 
 	for i, tg := range targetGroups {
+		if tg == nil {
+			err = fmt.Errorf("nil target group item found from: %s", filename)
```

Comment:
We already include the file name in the error log in the caller. Something like `errors.New("nil target group item found")` would be enough.